src/OctoshiftCLI.IntegrationTests/TestHelper.cs

```diff
@@ -0,0 +1,564 @@
+        }
+
+        private async Task<IEnumerable<string>> GetRepoCommitShas(string org, string repo)
+        {
+            var url = $"https://api.github.com/repos/{org}/{repo}/commits";
+            var commits = await _githubClient.GetAllAsync(url).ToListAsync();
+            return commits.Select(x => (string)x["sha"]).ToList();
+        }
+
+        private async Task<IEnumerable<(string id, string key, string url)>> GetAutolinks(string org, string repo)
+        {
+            var url = $"https://api.github.com/repos/{org}/{repo}/autolinks";
+            var autolinks = await _githubClient.GetAllAsync(url).ToListAsync();
+            return autolinks.Select(x => ((string)x["id"], (string)x["key_prefix"], (string)x["url_template"])).ToList();
+        }
+
+        private async Task<string> GetTeamIdPGroup(string org, string teamSlug)
+        {
+            var url = $"https://api.github.com/orgs/{org}/teams/{teamSlug}/external-groups";
+            var response = await _githubClient.GetAsync(url);
+            return (string)JObject.Parse(response)["groups"].Single()["group_name"];
+        }
+
+        private async Task<string> GetTeamRepoRole(string org, string team, string repo)
+        {
+            var url = $"https://api.github.com/orgs/{org}/teams/{team}/repos";
+            var response = await _githubClient.GetAllAsync(url).ToListAsync();
+            return (string)response.Single(x => (string)x["name"] == repo)["role_name"];
+        }
+
+        public string GetOsName()
+        {
+            return RuntimeInformation.IsOSPlatform(OSPlatform.Linux)
+                ? "linux"
+                : RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+                ? "windows"
+                : RuntimeInformation.IsOSPlatform(OSPlatform.OSX)
+                ? "macos"
+                : throw new InvalidOperationException("Could not determine OS");
+        }
+
+        public void RunCliMigration(string generateScriptCommand, string cliName, IDictionary<string, string> tokens)
+        {
+            var startInfo = new ProcessStartInfo
+            {
+                WorkingDirectory = GetOsDistPath(),
+                FileName = $"{cliName}",
+                Arguments = generateScriptCommand
+            };
+
+            if (tokens != null)
+            {
+                foreach (var token in tokens)
+                {
+                    if (startInfo.EnvironmentVariables.ContainsKey(token.Key))
+                    {
+                        startInfo.EnvironmentVariables[token.Key] = token.Value;
+                    }
+                    else
+                    {
+                        startInfo.EnvironmentVariables.Add("ADO_PAT", token.Value);
+                    }
+                }
+            }
+
+            _output.WriteLine($"Running command: {startInfo.FileName} {startInfo.Arguments}");
+            var p = Process.Start(startInfo);
+            p.WaitForExit();
```

Comment:
We can asynchronously wait for exit: [p.WaitForExitAsync()](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.waitforexitasync?view=net-6.0#system-diagnostics-process-waitforexitasync(system-threading-cancellationtoken)). That way the method will be `async` and non-blocking.

Reply:
done